core/src/main/java/edu/ucr/cs/riple/core/Annotator.java

```diff
@@ -93,34 +99,41 @@ private void explore() {
         new FieldDeclarationAnalysis(config.dir.resolve("class_info.tsv"));
     while (true) {
       Utility.buildProject(config);
-      Set<Fix> remainingFixes =
+      Stream<Fix> remainingFixes =
           Utility.readFixesFromOutputDirectory(
               config, Fix.factory(config, fieldDeclarationAnalysis));
       if (config.useCache) {
-        remainingFixes =
-            remainingFixes.stream()
-                .filter(fix -> !reports.containsKey(fix))
-                .collect(Collectors.toSet());
+        remainingFixes = remainingFixes.filter(fix -> !reports.containsKey(fix));
       }
-      ImmutableSet<Fix> fixes = ImmutableSet.copyOf(remainingFixes);
+      ImmutableSet<Fix> fixes = remainingFixes.collect(ImmutableSet.toImmutableSet());
       Bank<Error> errorBank = new Bank<>(config.dir.resolve("errors.tsv"), Error::new);
       Bank<Fix> fixBank =
           new Bank<>(
               config.dir.resolve("fixes.tsv"), Fix.factory(config, fieldDeclarationAnalysis));
       MethodInheritanceTree tree =
           new MethodInheritanceTree(config.dir.resolve(Serializer.METHOD_INFO_FILE_NAME));
-      RegionTracker tracker = new CompoundTracker(config.dir, tree);
+      RegionTracker tracker = new CompoundTracker(config, tree);
       Explorer explorer =
           config.exhaustiveSearch
               ? new ExhaustiveExplorer(injector, errorBank, fixBank, fixes, tree, config)
               : config.optimized
                   ? new OptimizedExplorer(
-                      injector, errorBank, fixBank, tracker, fixes, tree, config)
+                      injector, errorBank, fixBank, tracker, fixes, tree, config.depth, config)
```

Comment:
Since we are doing the analysis on downstream modules to only depth 1 and is different from the actual target module. Please let me know if you prefer changing this.

Reply:
Ah, I see, `config.depth` is passed here, but in `DownStreamDependencyAnalyzer.java` we are passing `1`. Makes sense.